### PR TITLE
Remove field accessors for `TxInWitness`

### PR DESCRIPTION
**Pull request by Jimbo4350**
## Description

Remove field accessors for `TxInWitness`

## Linked issue

https://iohk.myjetbrains.com/youtrack/issue/CDEC-302

## Type of change

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🛠 New feature (non-breaking change which adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [ ] 🔨 New or improved tests for existing code
- [ ] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [ ] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [ ] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [ ] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.

## Testing checklist

- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: client/src/Pos/Client/Txp/Util.hs [hunk 1]
```diff
@@ -243,10 +243,8 @@ makeMPubKeyTx
 makeMPubKeyTx pm getSs = makeAbstractTx mkWit
   where mkWit addr sigData =
           getSs addr <&> \ss ->
-              PkWitness
-              { twKey = safeToPublic ss
-              , twSig = safeSign pm SignTx ss sigData
-              }
+              PkWitness (safeToPublic ss) (safeSign pm SignTx ss sigData)
+
```
Comment (intricate) [inline client/src/Pos/Client/Txp/Util.hs]: Unnecessary newline here

File: client/src/Pos/Client/Txp/Util.hs [hunk 2]
```diff
@@ -279,10 +277,9 @@ makeMOfNTx
 makeMOfNTx pm validator sks txInputs txOutputs = either absurd identity $
     makeAbstractTx mkWit (map ((), ) txInputs) txOutputs
   where
-    mkWit _ sigData = Right $ ScriptWitness
-            { twValidator = validator
-            , twRedeemer = multisigRedeemer pm sigData sks
-            }
+    mkWit _ sigData =
+        Right $ ScriptWitness validator (multisigRedeemer pm sigData sks)
+
```
Comment (intricate) [inline client/src/Pos/Client/Txp/Util.hs]: Unnecessary newline here

File: client/src/Pos/Client/Txp/Util.hs [hunk 3]
```diff
@@ -293,10 +290,9 @@ makeRedemptionTx
 makeRedemptionTx pm rsk txInputs txOutputs = either absurd identity $
     makeAbstractTx mkWit (map ((), ) txInputs) txOutputs
   where rpk = redeemToPublic rsk
-        mkWit _ sigData = Right $ RedeemWitness
-            { twRedeemKey = rpk
-            , twRedeemSig = redeemSign pm SignRedeemTx rsk sigData
-            }
+        mkWit _ sigData =
+            Right $ RedeemWitness rpk (redeemSign pm SignRedeemTx rsk sigData)
+
```
Comment (intricate) [inline client/src/Pos/Client/Txp/Util.hs]: Unnecessary newline here

File: core/src/Pos/Core/Txp/TxWitness.hs [hunk 4]
```diff
@@ -28,12 +28,12 @@ type TxWitness = Vector TxInWitness
...
-    | ScriptWitness { twValidator :: !Script
-                    , twRedeemer  :: !Script }
-    | RedeemWitness { twRedeemKey :: !RedeemPublicKey
-                    , twRedeemSig :: !(RedeemSignature TxSigData) }
+    -- PkWitness twKey twSig
+    = PkWitness PublicKey TxSig
+    -- ScriptWitness twValidator twRedeemer
+    | ScriptWitness Script Script
+    -- RedeemWitness twRedeemKey twRedeemSig
+    | RedeemWitness RedeemPublicKey (RedeemSignature TxSigData)
```
Comment (mhuesch) [inline core/src/Pos/Core/Txp/TxWitness.hs]: Some strictness annotations were lost here - it's recommended to keep those: https://github.com/serokell/serokell-util/blob/master/serokell-style.md#dealing-with-laziness

Comment (Jimbo4350) [reply core/src/Pos/Core/Txp/TxWitness.hs]: Whoops, thanks @mhuesch

**Comment by erikd** [comment]
I kicked off another AppVeyor build. Hopefully it passes this time.

**Pull request merged**